### PR TITLE
Add helper text for product variations in Product images screen

### LINKED ISSUE #3933: Variations: Add image helper text
It's not very clear that variable products can have only one image, to help with possible confusion
, let's add a helper above the photo list screen.

Current | Expected(Android design)
--- | ---
ios-no-text | android-text

### PR DESCRIPTION
**Pull request by itsmeichigo**
Closes #3933 

## Description
This PR updates the Product images screen by adding a helper text about the image list of product variations. This means the Product images screen should stay intact for products that allow multiple photos.

| Normal products | Product variations |
| ------- | ------- |
|  |  |

## Solution
- Embed the content of top container view (the view containing top button and separator view) inside a vertical stack view. This makes it easier to show / hide the helper text.
- Insert the helper view below the separator view. This is basically a UIView containing a UILabel that allows for adding margins around the label.
- Configure the helper view by checking conditions: (1) whether the product does not allow multiple images and (2) the product is of type variable. If these 2 conditions are met, the helper text should be visible.

## Discussion
Since the helper text content is now specific to product variation, I had to check whether product is of type variable. If in the future the text is more generic, the check for allowing multiple images is sufficient.

## Testing
This can be checked by visiting the Product images through different flows:
- When editing products, select any image in the list or choose to add new image in the main edit screen. The resulting image list screen should not show the helper text.
- Create or edit a variable product, choose a variation to add or edit its photo. The helper text should be shown in this case. 

Update release notes:
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Review by ealeksandrov — APPROVED** [review]
LGTM :shipit:

**Review by ealeksandrov — APPROVED** [review]
Still LGTM :shipit:

**Comment by Ecarrion** [comment]
@woocommerce/mobile-designers We only have an android design for this change.

android-text

Could you confirm if the font/style @itsmeichigo is proposing is ok? or if you would want something different?

Screen Shot 2021-04-28 at 10 20 58 AM

Thanks!

**Review by Ecarrion — APPROVED** [review]
Changes look good to me too! 🎉 

Sorry @ealeksandrov for dismissing your review. I miss-tapped the button when trying to request a review from the design team! 🤦 

Let's wait on their opinion before merging!

**Review by Garance91540 — COMMENTED** [review]
Hi there! One small change request: can we make the text size iOS/Footnote?

Thanks! https://d.pr/i/BpPM4G

**Comment by itsmeichigo** [comment]
Thank you @Garance91540 - below is the screenshot of the screen after the font update. However, the copy is different from your design, should I update that as well?

**Comment by Garance91540** [comment]
Yes if possible let's update the copy too, and the right and left padding (should be 16px). Thanks!

**Pull request merged**
